extensions/wikia/WikiFactory/Close/maintenance.php

```diff
@@ -416,6 +413,38 @@ private function getDirTree( $dir ) {
 	}
 
 	/**
+	 * Remove DFS bucket of a given wiki
+	 *
+	 * @see PLATFORM-1700
+	 * @param int $cityId
+	 */
+	private function removeBucket( $cityid ) {
+		try {
+			$swift = \Wikia\SwiftStorage::newFromWiki( $cityid );
+			$this->log( sprintf( "Removing DFS bucket /%s%s", $swift->getContainerName(), $swift->getPathPrefix() ) );
+
+			// s3cmd --recursive del s3://BUCKET/OBJECT / Recursively delete files from bucket
+			$cmd = sprintf(
+				'sudo /usr/bin/s3cmd -c %s --recursive del s3://%s%s/',
```

Comment:
Unfortunately, CloudFiles library (from Rackspace) that we use in MediaWiki only support removing single files and whole buckets.

Can't think of something different then fetching the list of ALL files in the bucket (via CloudFiles API call), filtering it and removing.

And we already use `s3cmd` in this script for creating images backups.